rustls/src/vecbuf.rs

```diff
@@ -45,11 +45,10 @@ impl ChunkVecBuffer {
 
     /// How many bytes we're storing
     pub(crate) fn len(&self) -> usize {
-        let mut len = 0;
-        for ch in &self.chunks {
-            len += ch.len();
-        }
-        len
+        self.chunks
+            .iter()
+            .map(|ch| ch.len())
+            .sum()
```

Comment:
Does this have a performance benefit? Seems surprising.

Reply:
Reworded commit message.

Reply:
It doesn't look much more idiomatic to me, but it's one line less so I'll take it. 👍